test/bdd/pkg/issuecredential/issuecredential_sdk_steps.go

```diff
@@ -0,0 +1,258 @@
+	bddContext *context.BDDContext
+	clients    map[string]*issuecredential.Client
+	actions    map[string]chan service.DIDCommAction
+	events     map[string]chan service.StateMsg
+}
+
+// NewIssueCredentialSDKSteps creates steps for the issuecredential with SDK
+func NewIssueCredentialSDKSteps(ctx *context.BDDContext) *SDKSteps {
+	return &SDKSteps{
+		bddContext: ctx,
+		clients:    make(map[string]*issuecredential.Client),
+		actions:    make(map[string]chan service.DIDCommAction),
+		events:     make(map[string]chan service.StateMsg),
+	}
+}
+
+// RegisterSteps registers agent steps
+func (a *SDKSteps) RegisterSteps(s *godog.Suite) {
+	s.Step(`^"([^"]*)" requests credential from "([^"]*)"$`, a.sendsRequest)
+	s.Step(`^"([^"]*)" accepts request and sends credential to the Holder$`, a.acceptRequest)
+	s.Step(`^"([^"]*)" sends proposal credential to the "([^"]*)"`, a.sendsProposal)
+	s.Step(`^"([^"]*)" accepts a proposal and sends an offer to the Holder`, a.acceptProposal)
+	s.Step(`^"([^"]*)" sends an offer to the "([^"]*)"$`, a.sendsOffer)
+	s.Step(`^"([^"]*)" accepts an offer and sends a request to the Issuer`, a.acceptOffer)
+	s.Step(`^"([^"]*)" accepts credential with name "([^"]*)"$`, a.acceptCredential)
+	s.Step(`^"([^"]*)" checks that credential is being stored under "([^"]*)" name$`, a.checkCredential)
+}
+
+func (a *SDKSteps) checkCredential(agent, name string) error {
+	return retry(retryAttempts, time.Second, func() error {
+		store, err := verifiableStore.New(a.bddContext.AgentCtx[agent])
+		if err != nil {
+			return err
+		}
+
+		ID, err := store.GetCredentialIDByName(name)
+		if err != nil {
+			return err
+		}
+
+		_, err = store.GetCredential(ID)
+		return err
+	})
+}
+
+func retry(attempts int, sleep time.Duration, fn func() error) error {
+	if err := fn(); err != nil {
+		attempts--
+		if attempts > 0 {
+			time.Sleep(sleep)
+			return retry(attempts, sleep, fn)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (a *SDKSteps) sendsOffer(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent1].SendOffer(&protocol.OfferCredential{}, conn.MyDID, conn.TheirDID)
+}
+
+func (a *SDKSteps) sendsProposal(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent1].SendProposal(&protocol.ProposeCredential{}, conn.MyDID, conn.TheirDID)
+}
+
+func (a *SDKSteps) sendsRequest(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
```

Comment:
lines 117-128, 134-145 and 151-162 are similar. These can be moved to a common function (init clients and getting the connection).

Reply:
Done!